sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -91,22 +90,34 @@ public boolean start() throws IOException {
     consumer = spec.getConsumerFactoryFn().apply(spec.getConsumerConfig());
     consumerSpEL.evaluateAssign(consumer, spec.getTopicPartitions());
 
+    String schemaRegistryURL =
+        (String)
+            spec.getConsumerConfig().get(KafkaAvroDeserializerConfig.SCHEMA_REGISTRY_URL_CONFIG);
+    SchemaRegistryClient registryClient =
+        KafkaIO.Read.getSchemaRegistryClient(
+            schemaRegistryURL, spec.getSchemaRegistryClientFactoryFn());
+
     try {
       if (spec.getKeyDeserializer().equals(KafkaAvroDeserializer.class)) {
-        SchemaRegistryClient registryClient = getSchemaRegistryClient(spec);
+        if (registryClient == null) {
```

Comment:
Remove validation per above comment on checkArgument, if we cannot build a client we fail.